src/serialization.jl

```diff
@@ -21,36 +31,113 @@ mutable struct TrialPreV006
+            "convert existing saved benchmarks to JSON in order to deserialize them with\n" *
+            "this version of BenchmarkTools."
+        else
+            "Only JSON deserialization is supported. Use `load(\"$noext.json\", args...)`."
+        end
+        throw(ArgumentError(msg))
+    end
+    if !isempty(args)
+        throw(ArgumentError("Looking up deserialized values by name is no longer supported, " *
+                            "as names are no longer saved."))
     end
-    return nothing
+    parsed = open(JSON.parse, filename, "r")
+    if !isa(parsed, Vector) || length(parsed) != 2 || !isa(parsed[1], Dict) || !isa(parsed[2], Vector)
+        error("Unexpected JSON format. Was this file produced by BenchmarkTools?")
+    end
+    versions = parsed[1]::Dict
+    values = parsed[2]::Vector
+    map!(recover, values, values)
 end
 
-@inline function load(filename, args...)
-    # no version-based rules are needed for now, we just need
-    # to check that version information exists in the file.
-    if JLD.jldopen(file -> JLD.exists(file, VERSION_KEY), filename, "r")
-        result = JLD.load(filename, args...)
-    else
-        JLD.translate("BenchmarkTools.Parameters", "BenchmarkTools.ParametersPreV006")
-        JLD.translate("BenchmarkTools.Trial", "BenchmarkTools.TrialPreV006")
-        result = JLD.load(filename, args...)
-        JLD.translate("BenchmarkTools.Parameters", "BenchmarkTools.Parameters")
-        JLD.translate("BenchmarkTools.Trial", "BenchmarkTools.Trial")
-    end
-    return result
-end
+#function JLD.readas(p::ParametersPreV006)
+#    return Parameters(p.seconds, p.samples, p.evals, Float64(p.overhead), p.gctrial,
+#                      p.gcsample, p.time_tolerance, p.memory_tolerance)
+#end
+#
+#function JLD.readas(t::TrialPreV006)
+#    new_times = convert(Vector{Float64}, t.times)
+#    new_gctimes = convert(Vector{Float64}, t.gctimes)
+#    return Trial(t.params, new_times, new_gctimes, t.memory, t.allocs)
+#end
+#
+#function save(filename, args...)
+#    JLD.save(filename, VERSION_KEY, VERSIONS, args...)
+#    JLD.jldopen(filename, "r+") do io
+#        JLD.addrequire(io, BenchmarkTools)
+#    end
+#    return nothing
+#end
+#
+#@inline function load(filename, args...)
+#    # no version-based rules are needed for now, we just need
+#    # to check that version information exists in the file.
+#    if JLD.jldopen(file -> JLD.exists(file, VERSION_KEY), filename, "r")
+#        result = JLD.load(filename, args...)
+#    else
+#        JLD.translate("BenchmarkTools.Parameters", "BenchmarkTools.ParametersPreV006")
+#        JLD.translate("BenchmarkTools.Trial", "BenchmarkTools.TrialPreV006")
+#        result = JLD.load(filename, args...)
+#        JLD.translate("BenchmarkTools.Parameters", "BenchmarkTools.Parameters")
+#        JLD.translate("BenchmarkTools.Trial", "BenchmarkTools.Trial")
+#    end
+#    return result
+#end
```

Comment:
I'm sure you'd do this eventually anyway, but just to be explicit, we should definitely delete this commented out stuff before merging...what a joy it will be to be rid of it!

Reply:
Yeah definitely